x-pack/plugins/infra/public/pages/metrics/hosts/components/hosts_table.tsx

```diff
@@ -30,36 +29,29 @@ const HOST_METRICS: Array<{ type: SnapshotMetricType }> = [
 ];
 
 export const HostsTable = () => {
-  const { sourceId } = useSourceContext();
-  const { buildQuery, dateRangeTimestamp } = useUnifiedSearchContext();
+  const { baseRequest, setHostViewState, hostViewState } = useHostsViewContext();
+  const { onSubmit } = useUnifiedSearchContext();
   const [properties, setProperties] = useTableProperties();
 
-  const timeRange: InfraTimerangeInput = {
-    from: dateRangeTimestamp.from,
-    to: dateRangeTimestamp.to,
-    interval: '1m',
-    ignoreLookback: true,
-  };
-
-  const esQuery = buildQuery();
-
   // Snapshot endpoint internally uses the indices stored in source.configuration.metricAlias.
   // For the Unified Search, we create a data view, which for now will be built off of source.configuration.metricAlias too
   // if we introduce data view selection, we'll have to change this hook and the endpoint to accept a new parameter for the indices
-  const { loading, nodes, reload } = useSnapshot({
-    filterQuery: esQuery ? JSON.stringify(esQuery) : null,
-    metrics: HOST_METRICS,
-    groupBy: [],
-    nodeType: 'host',
-    sourceId,
-    currentTime: dateRangeTimestamp.to,
-    accountId: '',
-    region: '',
-    timerange: timeRange,
-    includeTimeseries: false,
+  const { loading, nodes, error } = useSnapshot({
+    ...baseRequest,
+    metrics: HOST_TABLE_METRICS,
   });
 
-  const items = useHostTable(nodes);
+  useEffect(() => {
+    if (hostViewState.loading !== loading) {
+      setHostViewState({
+        loading,
+        totalHits: nodes.length,
+        error,
+      });
+    }
+  }, [error, loading, nodes.length, hostViewState, setHostViewState]);
```

Comment:
Passing the whole `hostViewState` as a dependency, this effect will run each time it changes anything in the state, should this dep be `hostViewState.loading` instead?

Reply:
True. I forgot to update it after some changes. Thanks